frontend/src/layout/components/Sidebar/index.vue

```diff
@@ -134,18 +134,19 @@ function getCheckedLabels(json: Node): string[] {
 const search = async () => {
     await checkIsSystemIntl();
     let checkedLabels: any[] = [];
-    if (!globalStore.isIntl) {
-        const res = await getSettingInfo();
-        const json: Node = JSON.parse(res.data.xpackHideMenu);
-        checkedLabels = getCheckedLabels(json);
-    }
+    const res = await getSettingInfo();
+    const json: Node = JSON.parse(res.data.xpackHideMenu);
+    checkedLabels = getCheckedLabels(json);
 
     let rstMenuList: RouteRecordRaw[] = [];
     menuStore.menuList.forEach((item) => {
         let menuItem = JSON.parse(JSON.stringify(item));
         let menuChildren: RouteRecordRaw[] = [];
         if (menuItem.path === '/xpack') {
             if (checkedLabels.length) {
+                menuItem.children = menuItem.children.filter((child: any) => {
+                    return !(globalStore.isIntl && child.path.includes('/xpack/alert'));
+                });
                 menuItem.children.forEach((child: any) => {
                     for (const str of checkedLabels) {
                         if (child.name === str) {
```

Comment:
The provided code appears to be part of a Vue.js application using the Element Plus UI library. Here's an assessment of the code:

### Potential Issue
The condition `if (menuItem.path === '/xpack')` is inside the loop that processes each MenuItem within the MenuList. This means it will process all children regardless of whether their path starts with `/xpack/`. If there are children at other paths under `/xpack`, they may become inaccessible depending on the logic used later.

### Suggestion
Consider refactoring this loop structure so that you only apply the filtering logic when a specific condition (e.g., checking for sub-paths beneath `/xpack`) occurs, rather than processing all items every time.

Here’s one possible way to refactor it:

```typescript
// Initialize filtered menu list directly outside the main loop
let filteredMenuList: RouteRecordRaw[] = [];

menuStore.menuList.forEach((item) => {
    // Process parent item
    let menuItem = JSON.parse(JSON.stringify(item));
    if (menuItem.path === "/xpack") {
        // Determine which parts should remain visible based on user settings
        filteredMenuList.push(menuItem); // Add current item since it matches '/xpack'
        
        // Apply visibility filter to child menu options
        menuItem.children = menuItem.children.filter(child => !child.path.includes("/xpack/alert"));
        
        // Recursively process the child menu
        recursivelyFilterChildMenuItems(filteredMenuList, child);
    } else {
        // Directly add non-menu-item nodes to final result set.
        rstMenuList.push(menuItem);
    }
});

function recursivelyFilterChildMenuItems(parentList: RouteRecordRaw[], targetMenuItem: RouteRecordRaw) {
    // ... Existing code remains same ...
}
```

By doing this, you ensure that each MenuItem is processed only once, improving efficiency when dealing with nested structures.

---

Make sure this change aligns fully with your application's requirements and architecture before implementing it!